### PR TITLE
EPMRPP-37400/EPMRPP-36997/EPMRPP-36996 Admin part. Status page common logic.

### PR DESCRIPTION
**Pull request by AmsterGet**

### CODE REVIEW

File: app/src/common/urls.js [hunk 1]
```diff
@@ -97,6 +100,11 @@ export const URLS = {
     `${urlBase}project/${activeProject}/usernames?filter.cnt.users=`,
   projectIndex: (activeProject) => `${urlBase}project/${activeProject}/index`,
 
+  projectStatus: (activeProject, interval) =>
+    `${urlBase}project/list/${activeProject}/${getQueryParams({
```
Comment (DiscoElevator) [inline app/src/common/urls.js]: `${activeProject}${getQueryParams({` - slash is not needed for query parameters

File: app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    interval: PropTypes.string,
+  };
+
+  static defaultProps = {
+    interval: '',
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
```
Comment (DiscoElevator) [inline app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx]: You may define default state as a class field, so you won't need constructor.

File: app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    this.setState({
+      loading: true,
+    });
+    const widgetsWithUrls = statusPageWidgets.filter((item) => item.getUrl);
+    widgetsWithUrls.push(activityItem);
+
+    Promise.all(widgetsWithUrls.map((item) => fetch(item.getUrl(projectId, interval))))
+      .then((responses) => {
+        const itemsData = {};
+        widgetsWithUrls.forEach((item, index) => {
```
Comment (DiscoElevator) [inline app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx]:
```
const itemsData = widgetsWithUrls.reduce((acc, item, index) => ({...acc, [item.id]: responses[index]}), {})
```

File: app/src/pages/admin/projectStatusPage/statusPageContent/statusPageWidgets.jsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import { WIDGETS_IDS } from '../constants';
+import { messages } from './messages';
+
+export const statusPageWidgets = [
+  {
+    title: messages.general,
+    id: WIDGETS_IDS.general,
+    source: WIDGETS_IDS.general,
+    component: (data) => <GeneralInfo data={data} />,
+    getUrl: (...params) => URLS.projectStatus(...params),
```
Comment (DiscoElevator) [inline app/src/pages/admin/projectStatusPage/statusPageContent/statusPageWidgets.jsx]: `getUrl: URLS.projectStatus`

File: app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  componentDidUpdate(prevProps) {
+    if (
+      prevProps.interval !== this.props.interval ||
+      prevProps.projectId !== this.props.projectId
+    ) {
+      this.fetchData();
+    }
+  }
+
+  getItem = (item) => (
```
Comment (DiscoElevator) [inline app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx]: `renderWidget`

File: app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      {this.state.loading ? <SpinningPreloader /> : item.component(this.state[item.source])}
+    </StatusPageItem>
+  );
+
+  fetchData = () => {
+    const { projectId, interval } = this.props;
+    this.setState({
+      loading: true,
+    });
+    const widgetsWithUrls = statusPageWidgets.filter((item) => item.getUrl);
```
Comment (DiscoElevator) [inline app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx]:
```
const widgetsWithUrls = [
  ...statusPageWidgets.filter((item) => item.getUrl),
  activityItem
];
```

File: app/src/common/urls.js (hunk 1 above)
Comment (AmsterGet) [reply app/src/common/urls.js]: Fixed

File: app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx (hunk 3 above)
Comment (AmsterGet) [reply app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx]: Fixed

File: app/src/pages/admin/projectStatusPage/statusPageContent/statusPageWidgets.jsx (hunk 4 above)
Comment (AmsterGet) [reply app/src/pages/admin/projectStatusPage/statusPageContent/statusPageWidgets.jsx]: Fixed

File: app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx (hunk 5 above)
Comment (AmsterGet) [reply app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx]: Fixed

File: app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx (hunk 6 above)
Comment (AmsterGet) [reply app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx]: Fixed

File: app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx (hunk 2 above)
Comment (AmsterGet) [reply app/src/pages/admin/projectStatusPage/statusPageContent/statusPageContent.jsx]: Fixed

**Pull request merged**